cookbooks/bcpc_kafka/recipes/kafka.rb

```diff
@@ -107,9 +107,69 @@
 node.default[:kafka][:generic_opts] +=
   " -Djute.maxbuffer=#{node[:bcpc][:hadoop][:jute][:maxbuffer] rescue 0xfffff}"
 
-# Increase the default FD limit -- kafka opens a lot of sockets.
+# Increase the default FD limit -- Kafka opens a lot of sockets.
 node.default[:kafka][:ulimit_file] = 32_768
 
+#
+# This is the canned setup from bcpc-hadoop for Kerberos. The
+# implementation is smeared across bcpc-hadoop, bcpc, and bach_krb5.
+#
+# Scroll down for the kafka-specific material.
+#
+include_recipe 'bach_krb5::keytab_directory'
+include_recipe 'bach_krb5::krb5_client'
+
+#
+# Pre-create the Kafka user.  This is normally done by the upstream
+# kafka cookbook, but Kerberos configuration requires the keytab owner
+# to exist before it can proceed.
+#
+user 'dummy_kafka_user' do
+  action :create
+  username 'kafka'
+  home '/var/empty/kafka'
+  shell '/sbin/nologin'
+  #
+  # Chef does not have a :create_if_missing for the user resource, so
+  # we avoid trying to alter an existing user by refusing to converge
+  # if a kafka user exists.
+  #
+  not_if 'id kafka'
```

Comment:
Though `id` is more concise we have used `getent` elsewhere (e.g. [here](https://github.com/bloomberg/chef-bach/blob/3be0bf618c2cb195c10ff403f858bdb58b13249a/cookbooks/bcpc/recipes/nscd.rb#L40)).